include/measurement_kit/common/measure_speed.hpp

```diff
@@ -0,0 +1,54 @@
+// Part of measurement-kit <https://measurement-kit.github.io/>.
+// Measurement-kit is free software. See AUTHORS and LICENSE for more
+// information on the copying conditions.
+#ifndef MEASUREMENT_KIT_COMMON_MEASURE_SPEED_HPP
+#define MEASUREMENT_KIT_COMMON_MEASURE_SPEED_HPP
+
+#include <measurement_kit/common/callback.hpp>
+#include <measurement_kit/common/utils.hpp>
+
+namespace mk {
+
+class MeasureSpeed {
+  public:
+    const double start_time = mk::time_now();
+    double previous = start_time;
+    double snap_delay = -1.0;
+    size_t total = 0;
+
+    MeasureSpeed(double delay) {
+        snap_delay = delay;
+    }
+    MeasureSpeed() {}
+
+    double speed(double ct) const {
+        double elapsed = ct - previous;
+        if (elapsed <= 0.0) {
+            return 0.0;
+        }
+        return (total * 8.0) / 1000.0 / elapsed;
```

Comment:
If this class is meant to be general class used by many tests, maybe different tests want to measure the speed in a different way and not using this function, so I confirm that this class should be in the ndt folder!